staging/src/k8s.io/apimachinery/pkg/util/wait/wait.go

```diff
@@ -317,103 +337,46 @@ func (b *Backoff) Step() time.Duration {
 	return duration
 }
 
-// BackoffManager manages backoff with a particular scheme based on its underlying implementation. It provides
-// an interface to return a timer for backoff, and caller shall backoff until Timer.C() drains. If the second Backoff()
-// is called before the timer from the first Backoff() call finishes, the first timer will NOT be drained and result in
-// undetermined behavior.
-// The BackoffManager is supposed to be called in a single-threaded environment.
-type BackoffManager interface {
-	Backoff() clock.Timer
-}
-
-type exponentialBackoffManagerImpl struct {
-	backoff              *Backoff
-	backoffTimer         clock.Timer
-	lastBackoffStart     time.Time
-	initialBackoff       time.Duration
-	backoffResetDuration time.Duration
-	clock                clock.Clock
-}
-
-// NewExponentialBackoffManager returns a manager for managing exponential backoff. Each backoff is jittered and
-// backoff will not exceed the given max. If the backoff is not called within resetDuration, the backoff is reset.
-// This backoff manager is used to reduce load during upstream unhealthiness.
-func NewExponentialBackoffManager(initBackoff, maxBackoff, resetDuration time.Duration, backoffFactor, jitter float64, c clock.Clock) BackoffManager {
```

Comment:
this change affects a lot of projects https://grep.app/search?q=NewExponentialBackoffManager, can't we keep both constructor and change only the internals?

Reply:
Let me review (using https://grep.app/search?q=wait.NewExponentialBackoffManager&case=true&words=true)

The signature change is roughly:

```
- bm := wait.NewExponentialBackoffManager(time.Millisecond, maxBackoff, 100*time.Millisecond, 2.0, 1.0, fakeClock)
+ bm := wait.Backoff{Duration: time.Millisecond, Cap: maxBackoff, Factor: 2.0, Jitter: 1.0, Steps: math.MaxInt}.DelayWithReset(fakeClock, 100*time.Millisecond)
```

and then at point of use you call `.Timer(<your clock|wait.RealTimer>)` OR you invoke the delay func.  I am slightly worried about the mechanical change (people have to get the right arguments to Backoff) but the impact to point of use is low.

That being said there is no reason we have to remove the constructor inside the codebase - we can mark it deprecated and take feedback.  I'll make that change, thanks.

Reply:
See my other question about whether to have a helper on the existing manager and delay when they have to get rid of the manager.  I suspect the answer should be "unless we remove the manager constructor now, just add the helper to bridge the gap".

Reply:
(which is what i implemented)